### ISSUE TITLE
Compress-Archive not including folders starting with . on Ubuntu

### ISSUE THREAD

**Issue by Agazoth** [PowerShell/PowerShell#24482]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

When compressing an archive containing a folder starting with . behaves differently on Windows and Ubuntu.

On Windows. the folder starting with . is included

On Ubuntu, the folder starting with . is not included

This is the test folder:
C:\TEMP\TESTZIP
│   File.txt
│
└───.Folder
        FileInFolder.txt

### Expected behavior

```console
Compress-Archive /mnt/c/temp/TestZip/* /mnt/c/temp/TestZipubuntu.zip -Force

# /mnt/c/temp/TestZipubuntu.zip contains the .Folder folder and the FileInFolder.txt file
```

### Actual behavior

```console
/mnt/c/temp/TestZipubuntu.zip only contains File.txt
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Ubuntu 22.04.4 LTS
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![{33D59379-B071-4DE8-9310-FF6C6639895A}](https://github.com/user-attachments/assets/951ddbf1-4656-485a-bdd3-5711f717ffc7)

![{31B94074-AA5E-409A-8D12-8537437C47C8}](https://github.com/user-attachments/assets/e31d0893-e8e8-413a-81db-3b6034ab24e4)

**Comment by rhubarb-geek-nz** [comment]
Indeed, this is even documented [Compress-Archive](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/compress-archive?view=powershell-7.4)

> The Compress-Archive cmdlet ignores hidden files and folders when creating or updating the archive file. On non-Windows machines, this includes files and folders with name that begins with the period (.) character.
> To ensure hidden files and folders are compressed into the archive, use the .NET API instead.

**Comment by Agazoth** [comment]
Hidden folders and folders starting with . are not the same thing in Windows. A hidden folder in Windows has the Hidden attribute set.

On Ubuntu, the . does indicate that the folder is hidden.

This leads to quite different behaviour in the cmdlet

**Comment by rhubarb-geek-nz** [comment]
This looks like a duplicate of [#8378](https://github.com/PowerShell/PowerShell/issues/8378), from 2018. This was marked as "Resolution External" which led to [#66](https://github.com/PowerShell/Microsoft.PowerShell.Archive/issues/66).
That [repository](https://github.com/PowerShell/Microsoft.PowerShell.Archive) is now archived.

**Comment by santisq** [comment]
@Agazoth In case you're looking for a workaround, [`Compress-ZipArchive`](https://github.com/santisq/PSCompression/blob/main/docs/en-US/Compress-ZipArchive.md) shouldn't have this problem 😉

**Comment by Agazoth** [comment]
@santisq awesome! Thanks 😊

**Comment by PedroC88** [comment]
I believe this is also an issue, while I see the logic behind it, it's still changing behavior. I have an Azure Function VS project that produces a folder called _.azurefunctions_ when built, and this is running on Azure Agents, when I run this on Linux and deploy, the deployment fails because the _.azurefunctions_ directory is missing in the artifacts.

At the very least the command should have an option to include those folders on demand, and the parameter can be ignored on Windows since they would be included by default.

**Comment by SaebAmini** [comment]
Wasted a couple of hours figuring out why my deployed web app isn't including the `.well-known` folder because of this.

**Comment by rhubarb-geek-nz** [comment]
... ah, a well-known problem!
